### PR TITLE
DM-7281: Update python testing dev guide to reflect updated TestCase methods

### PR DESCRIPTION
**Pull request by jonathansick**
Draft at https://developer.lsst.io/v/DM-7281/coding/python_testing.html

### CODE REVIEW

**Comment by jonathansick** [comment]
If you're (temporarily) switching out the mocked API ref links for URL links, you can drop the leading `~`. It's role is to remove the namespace in the published doc and show only the class/function name.

File: coding/python_testing.rst [hunk 1]
```diff
@@ -110,14 +110,18 @@ LSST Utility Test Support Classes
...
 
-Inheriting from :lclass:`lsst.utils.tests.TestCase` rather than :class:`unittest.TestCase` enables two new asserts that are useful for doing element-wise comparison of two :mod:`numpy` arrays.
+Inheriting from :lclass:`lsst.utils.tests.TestCase` rather than :class:`unittest.TestCase` enables two new asserts that are useful for doing element-wise comparison of two :mod:`numpy`-like arrays.
 
-:lmeth:`~lsst.utils.tests.TestCase.assertClose`
-   Asserts that the corresponding elements in two arrays are within the specified tolerance.
-:lmeth:`~lsst.utils.tests.TestCase.assertNotClose`
-   Asserts that the two arrays are different.
+`lsst.utils.tests.TestCase.assertFloatsAlmostEqual <https://lsst-web.ncsa.illinois.edu/doxygen/x_masterDoxyDoc/namespacelsst_1_1utils_1_1tests.html#a09ee2482a2e8d71e8612c0378f4286fc>`_
+   Asserts that floating point scalars and/or arrays are equal are within the specified tolerance. The default tolerance is significantly tighter than the tolerance used by `unittest.TestCase.assertAlmostEqual` or `numpy.testing.assert_almost_equal`; if you are replacing either of those methods you may have to specify `rtol` and/or `atol` to prevent failing asserts.
```
Comment (jonathansick) [inline coding/python_testing.rst]: Could you switch 1 sentence per line?

Comment (jonathansick) [inline coding/python_testing.rst]: You'll need to add a `:meth:` in from of the `assertAlmostEqual` and `assert_almost_equal` expressions in unittest and numpy.

File: coding/python_testing.rst [hunk 2]
```diff
@@ -110,14 +110,18 @@ LSST Utility Test Support Classes
...
+`lsst.utils.tests.TestCase.assertFloatsNotEqual <https://lsst-web.ncsa.illinois.edu/doxygen/x_masterDoxyDoc/namespacelsst_1_1utils_1_1tests.html#a4fc68518d134e3656499898653a3bce3>`_
+   Asserts that floating point scalars and/or arrays are not equal.
 
-Additionally, :lmod:`~lsst.afw` provides additional asserts that get loaded into :lclass:`lsst.utils.tests.TestCase` when the associated module is loaded.
+Note that :lmeth:`~lsst.utils.tests.TestCase.assertClose` and :lmeth:`~lsst.utils.tests.TestCase.assertNotClose` methods have been deprecated to the above methods.
+
+Additionally, :lmod:`~lsst.afw` provides additional asserts that get loaded into :lclass:`lsst.utils.tests.TestCase` when the associated module is loaded. These include methods for `Coords <https://lsst-web.ncsa.illinois.edu/doxygen/x_masterDoxyDoc/namespacelsst_1_1afw_1_1coord_1_1utils.html>`_, `Geom (Angles, Pairs, Boxes) <https://lsst-web.ncsa.illinois.edu/doxygen/x_masterDoxyDoc/namespacelsst_1_1afw_1_1geom_1_1utils.html>`_, and `Images <https://lsst-web.ncsa.illinois.edu/doxygen/x_masterDoxyDoc/namespacelsst_1_1afw_1_1image_1_1test_utils.html>`_, such as:
```
Comment (jonathansick) [inline coding/python_testing.rst]: One sentence per line.

File: coding/python_testing.rst (hunk 1 above)
Comment (jonathansick) [inline coding/python_testing.rst]: typo: `are equal are within`

Comment (jonathansick) [inline coding/python_testing.rst]: Put double backticks around `rtol` and `atol` to format them as generic code.

File: coding/python_testing.rst [hunk 3]
```diff
@@ -110,14 +110,18 @@ LSST Utility Test Support Classes
...
+`lsst.utils.tests.TestCase.assertFloatsEqual <https://lsst-web.ncsa.illinois.edu/doxygen/x_masterDoxyDoc/namespacelsst_1_1utils_1_1tests.html#a0e354bcea6ba8c11238882ede5058c03>`_
+   Asserts that floating point scalars and/or arrays are identically equal.
+`lsst.utils.tests.TestCase.assertFloatsNotEqual <https://lsst-web.ncsa.illinois.edu/doxygen/x_masterDoxyDoc/namespacelsst_1_1utils_1_1tests.html#a4fc68518d134e3656499898653a3bce3>`_
+   Asserts that floating point scalars and/or arrays are not equal.
 
-Additionally, :lmod:`~lsst.afw` provides additional asserts that get loaded into :lclass:`lsst.utils.tests.TestCase` when the associated module is loaded.
+Note that :lmeth:`~lsst.utils.tests.TestCase.assertClose` and :lmeth:`~lsst.utils.tests.TestCase.assertNotClose` methods have been deprecated to the above methods.
```
Comment (jonathansick) [inline coding/python_testing.rst]: → *by* the above methods

**Comment by jonathansick** [comment]
Thanks. Given the tiny fixes you can rebase+squash and merge. 👍

File: coding/python_testing.rst (hunk 1 above)
Comment (parejkoj) [inline coding/python_testing.rst]: All of these are done.

File: coding/python_testing.rst (hunk 2 above)
Comment (parejkoj) [inline coding/python_testing.rst]: Done.

File: coding/python_testing.rst (hunk 3 above)
Comment (parejkoj) [inline coding/python_testing.rst]: Done.

**Pull request merged**